File: gareth-api/src/main/java/org/craftsmenlabs/gareth/api/execution/ExecutionRunContext.java
package org.craftsmenlabs.gareth.api.execution;


public interface ExecutionRunContext {

    void storeString(String key, String value);

    void storeLong(String key, long value);

    void storeDouble(String key, double value);

    void storeBoolean(String key, boolean value);

    String getString(String key);

    long getLong(String key);

    double getDouble(String key);

    boolean getBoolean(String key);

    ExecutionResult toExecutionResult(ExecutionStatus running);
}

File: gareth-api/src/main/kotlin/org/craftsmenlabs/gareth/model/ExecutionRunContext.kt
package org.craftsmenlabs.gareth.model


interface  ExecutionRunContext {
    fun storeString(key: String, value: String)

    fun storeLong(key: String, value: Long)

    fun storeDouble(key: String, value: Double)

    fun storeBoolean(key: String, value: Boolean)

    fun getString(key: String): String

    fun getLong(key: String): Long

    fun getDouble(key: String): Double

    fun getBoolean(key: String): Boolean

    fun toExecutionResult(running: ExecutionStatus): ExecutionResult
}